packages/edit-site/src/components/page-pages/index.js

```diff
@@ -48,17 +50,39 @@ export default function PagePages() {
 		hiddenFields: [ 'date', 'featured-image' ],
 		layout: {},
 	} );
-	// Request post statuses to get the proper labels.
+
 	const { records: statuses } = useEntityRecords( 'root', 'status' );
-	const postStatuses = useMemo(
-		() =>
-			statuses === null
-				? EMPTY_OBJECT
-				: Object.fromEntries(
-						statuses.map( ( { slug, name } ) => [ slug, name ] )
-				  ),
-		[ statuses ]
-	);
+	const defaultStatuses = useMemo( () => {
+		return statuses === null
+			? DEFAULT_STATUSES
+			: statuses
+					.filter( ( { slug } ) => slug !== 'trash' )
+					.map( ( { slug } ) => slug )
+					.sort()
+					.join();
+	}, [ statuses ] );
+
+	useEffect( () => {
+		// Only update the view if the statuses received from the endpoint
+		// are different from the DEFAULT_STATUSES provided initially.
+		//
+		// The pages endpoint depends on the status endpoint via the status filter.
+		// Initially, this code filters the pages request by DEFAULT_STATUTES,
+		// instead of using the default (publish).
+		// https://developer.wordpress.org/rest-api/reference/pages/#list-pages
+		//
+		// By doing so, it avoids a second request to the pages endpoint
+		// upon receiving the statuses when they are the same (most common scenario).
+		if ( DEFAULT_STATUSES !== defaultStatuses ) {
+			setView( {
+				...view,
+				filters: {
+					...view.filters,
+					status: defaultStatuses,
+				},
+			} );
+		}
+	}, [ defaultStatuses ] );
```

Comment:
There's a few things here.

 - I don't think we should update the "view" programmatically. It will create an undo level... 
 - I don't think the initial view should depend on the REST API of the status. Can the filter be "empty" initially or fixed using a value we actually choose. If not, we shouldn't render anything and shouldn't compute the initial "view" object until the statuses are loaded (but I'd prefer to avoid this option if possible)

Reply:
> this effect will only set the view programmatically when the site has a different set of statuses from the default ones (very rare use case, though it can happen)

Not sure we should support that for the moment tbh.

Reply:
So, to confirm, you'd rather prevent the view from rendering until the statuses are loaded?

Reply:
> So, to confirm, you'd rather prevent the view from rendering until the statuses are loaded?

If we really want to support the themes that change the default statuses, yes. But maybe we don't care about this for now.

Reply:
I found a way to prevent us from programatically update the view https://github.com/WordPress/gutenberg/pull/55839